arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -896,7 +899,9 @@ Result RocksDBCollection::read(transaction::Methods* trx,
                                IndexIterator::DocumentCallback const& cb) const {
   TRI_IF_FAILURE("LogicalCollection::read") { return Result(TRI_ERROR_DEBUG); }
 
-  ::ReadTimeTracker timeTracker(_statistics._rocksdb_read_usec, _statistics);
+  if (_statistics._exportReadWriteMetrics) {
+    ::ReadTimeTracker timeTracker(_statistics._rocksdb_read_sec->get(), _statistics);
```

Comment:
This will not work. This creates a new block scope with the TimeTracker, which will directly be destroyed again as the block is left instantly. So this will **not** report the time the operation has taken, but only the time spent in the `if` block.

Reply:
Oh shit. You are correct